nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/rs2d/RS2DData.java

```diff
@@ -409,28 +410,113 @@ public String getFilePath() {
         return fpath;
     }
 
+    /**
+     * Get a parameter string value from its name.
+     * May include some implicit conversion for specific parameters (such as SPECTRAL_WIDTH when stored in ppm).
+     *
+     * @param name The parameter name
+     * @return the string representation of its value.
+     */
     @Override
     public String getPar(String name) {
+        Number convertedNumber = getConvertedNumberValue(name);
+        if (convertedNumber != null) {
+            return convertedNumber.toString();
+        }
+
+        List<? extends Number> convertedListNumber = getConvertedListNumberValue(name);
+        if (convertedListNumber != null) {
+            return convertedListNumber.toString();
+        }
+
         return header.get(name).stringValue();
     }
 
+    /**
+     * Get a parameter numeric value from its name.
+     * May include some implicit conversion for specific parameters (such as SPECTRAL_WIDTH when stored in ppm).
+     *
+     * @param name The parameter name
+     * @return the floating point representation of its value.
+     */
     @Override
     public Double getParDouble(String name) {
+        Number convertedNumber = getConvertedNumberValue(name);
+        if (convertedNumber != null) {
+            return convertedNumber.doubleValue();
+        }
```

Comment:
This was the original bug: python processing was changing sw from the parameter value based on its name, and this didn't include the ppm -> Hertz conversion, so the spectral width was badly interpreted.